drivers/api/no_os_eeprom.c

```diff
@@ -51,11 +51,17 @@
 int32_t no_os_eeprom_init(struct no_os_eeprom_desc **desc,
 			  const struct no_os_eeprom_init_param *param)
 {
-	if (!param)
-		return -EINVAL;
+	int32_t ret;
 
-	if ((param->platform_ops->init(desc, param)))
-		return -EINVAL;
+	if (!param || !param->platform_ops)
+		return -ENXIO;
```

Comment:
I think ENXIO is used for situations where you're searching for a device on a bus and cna't find it. EINVAL is more suitable here.